File: udemy/java/MediaLaco/src/mediaLaco/MediaLaco.java
package mediaLaco;
import javax.swing.JOptionPane;
public class MediaLaco {
	public static void main(String[] args) {
		float nota, soma, media;
		soma = 0;
		media = 0;
		JOptionPane.showMessageDialog(null,"informe as notas dos alunos");
		for(int i = 1; i<= 10; i++) {
			nota = Float.parseFloat(JOptionPane.showInputDialog("nota do "+i+"° aluno: "));
			if ((nota>=0) && (nota <=10)){
				soma = soma +nota;
				media++;
				
			}
			else {
				JOptionPane.showMessageDialog(null, "nota invalida, tente novamente ");
				i--;
			}
		}
		media = soma/media;
		JOptionPane.showMessageDialog(null, "A media da sala foi: "+ media);
		
	}

}

File: udemy/pascal/projetos/MediaLaco.pas
program MediaLaco;
var 
    soma, media, nota: real;
    p : integer;
begin
    soma := 0;
    writeln('informe as notas dos alunos da sala para tirar a media: '); 
    for p := 1 to 10 do
        begin
            read(nota);
            if (nota >=0) and (nota<=10) then
                soma := soma + nota
            else 
                begin
                    writeln('nota invalida, tente novamente');
                    p := p-1;
                end;
        end;
    media := soma/p;
    write('A media da sala eh: ',media:3:2);
end.
